codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithyjava/nameresolver/Native.java

```diff
@@ -144,21 +143,14 @@ public TypeName typeForShape(final ShapeId shapeId) {
     }
 
     return switch (shape.getType()) {
-      case BOOLEAN, BYTE, SHORT, INTEGER, LONG, FLOAT, DOUBLE -> {
-        /* From the Smithy Docs:
-         * "Boolean, byte, short, integer, long, float, and double shapes
-         * are only considered boxed if they are marked with the box trait.
-         * All other shapes are always considered boxed."
-         * https://smithy.io/1.0/spec/core/type-refinement-traits.html#box-trait
-         * While Smithy Models SHOULD use Smithy Prelude shapes to avoid this confusion,
-         * they do not have to.
-         * Hence, the need to check if these shapes have the box trait
-         */
-        final TypeName typeName = NATIVE_TYPES_BY_SIMPLE_SHAPE_TYPE.get(
-          shape.getType()
-        );
-        yield shape.hasTrait(BoxTrait.class) ? typeName.box() : typeName;
-      }
+      // All primitives are boxed for safety, even if @required
+      case BOOLEAN,
+        BYTE,
+        SHORT,
+        INTEGER,
+        LONG,
+        FLOAT,
+        DOUBLE -> NATIVE_TYPES_BY_SIMPLE_SHAPE_TYPE.get(shape.getType());
```

Comment:
I knew I had written logic to handle this!
If I only Smithy-2.0 had been published earlier... 
I literally quoted the Smithy-1.0 spec...